### PR TITLE
Feature/use del to remove

### PR DESCRIPTION
**Pull request by christianlundkvist**
Using the orbit-db command `db.del` to implement `remove`.

Also increased timeouts in the jest tests to 20s since the default 5s timeouts got triggered.

### CODE REVIEW

**Comment by christianlundkvist** [comment]
Failed in linting. Will update.

File: src/privateStore.js [hunk 1]
```diff
@@ -61,7 +64,13 @@ class PrivateStore {
...
    */
   async remove (key) {
-    return this.set(key, null)
+    if (!this.db) throw new Error('_sync must be called before interacting with the store')
+
+    const dbKey = this._genDbKey(key)
+    const hash = await this.db.del(dbKey)
+    this.updateRoot(hash)
+
+    return true
```
Comment (oed) [inline src/privateStore.js]: Isn't strictly part of this story, but can you change the last three lines to 
```js
return this.updateRoot(hash)
```
And please do the same thing for the `set` method :)

File: src/privateStore.js [hunk 2]
```diff
@@ -43,9 +43,12 @@ class PrivateStore {
   async set (key, value) {
     if (!this.db) throw new Error('_sync must be called before interacting with the store')
 
-    if (value != null) {
-      value = this._encryptEntry(value)
+    if (value == null) {
+      // Do nothing for null value
+      // To remove a value, call remove
+      return false
```
Comment (oed) [inline src/privateStore.js]: Do we really need to check this? I'd just remove it for simplicity. Not really any need to stop people from setting stuff to null.

**Comment by oed** [comment]
Thanks for fixing the timeout thing also! I think it was causing issues on circleci sometimes.

File: src/privateStore.js (hunk 2 above)
Comment (christianlundkvist) [reply src/privateStore.js]: Let me can leave the behavior the way it was regarding setting `null`.

File: src/privateStore.js (hunk 1 above)
Comment (christianlundkvist) [reply src/privateStore.js]: Yeah, this is nice. I'll make this change.

**Comment by christianlundkvist** [comment]
@oed Thanks for the comments! Updated.

File: src/privateStore.js [hunk 3]
```diff
@@ -46,12 +46,12 @@ class PrivateStore {
     if (value != null) {
       value = this._encryptEntry(value)
```
Comment (oed) [inline src/privateStore.js]: I don't get it, why should we check if the value is null? Can't we just always do the encryption regardless?

Comment (christianlundkvist) [reply src/privateStore.js]: So I'm not sure what encrypting null would mean? How is `null` represented as a `Uint8Array`? And when you decrypt, would you get back `null` or something else like the empty string?

Comment (oed) [reply src/privateStore.js]: Ok, that makes sense. I guess then we would need to think about if value is `undefined` also

Comment (christianlundkvist) [reply src/privateStore.js]: @oed Actually I verified that `(x != null)` returns `true` also if `x` is `undefined.` :smiley:

Comment (c-castillo) [reply src/privateStore.js]: @christianlundkvist `if (value)` works for both `null` and `undefined`. Javascript is weird sometimes.

Comment (christianlundkvist) [reply src/privateStore.js]: @oed Hmm that would mean that the value `false` would also trigger this. I’m actually wondering if we are even able to encrypt anything that’s not a string? In EIP 1024 I think they’re solving it by only encrypting/decrypting stringified JSON.

Comment (oed) [reply src/privateStore.js]: Right, didn't think of that. So you are saying that we just do
```js
value = JSON.stringify(value)
```
before encrypting, and
```js
value = JSON.parse(value)
```
after decrypting?

That makes sense. We can put this in the `_encryptEntry` and `_decryptEntry` functions.

Comment (christianlundkvist) [reply src/privateStore.js]: @oed Yes, I think that's a good way of solving it. Not sure how things like `undefined` are stringified, but this way we should be able to handle all the relevant cases.

**Comment by christianlundkvist** [comment]
@oed Updated encryption/decryption with `JSON.serialize()` and `JSON.parse()`.

**Pull request merged**